### ISSUE TITLE
Use concept of remote "path" rather than artifact "group"

### ISSUE THREAD

**Issue by benjamin-heasly** [isetbio/RemoteDataToolbox#10]
The API should use the general concept of directories or "paths" to organize artifacts, rather than the Maven-specific concept of "groups".

The API functions and documentation should reflect this change.

The directory concept will map cleanly onto the Maven group concept because groups may contain "." delimiters, similar to file separators.  See [naming conventions](https://maven.apache.org/guides/mini/guide-naming-conventions.html)

**Comment by benjamin-heasly** [comment]
I ran a test with Archiva and group names that contain "." delimiters.  I wanted to know if Archiva uses these delimiters as _just_ a nice syntax, or if it also interprets this syntax like folders and subfolders, with recursive artifact listing.

Short answer: yes.  Cool!

I made two groups: `first.middle.last` and `first.middle.flask`.  I put 4 artifacts into each group.

I asked Archiva to list artifacts in each group:

```
artifacts = rdtListArtifacts('first.middle.last', 'brainard-archiva')'
artifacts = rdtListArtifacts('first.middle.flask', 'brainard-archiva')'
```

Each time, the returned `artifacts` had size 4 -- no surprise there.

Then I asked Archiva to list artifacts in the "parent groups":

```
artifacts = rdtListArtifacts('first', 'brainard-archiva')'
artifacts = rdtListArtifacts('first.middle', 'brainard-archiva')'
```

Each time, the returned `artifacts` had size 8 -- nice!  Examining the artifact urls, I saw artifacts from both "sub-groups", `last` and `flask`.

This feature will make it relatively easy to treat Archiva "groups" like remote directories.
